kolibri/core/lessons/viewsets.py

```diff
@@ -97,3 +101,26 @@ def annotate_queryset(self, queryset):
         return annotate_array_aggregate(
             queryset, lesson_assignment_collections="lesson_assignments__collection"
         )
+
+    @action(detail=False)
+    def size(self, request, **kwargs):
+        lessons = self.get_queryset()
+        if lessons:
+            lessons_nodes_list = [{lesson.id: lesson.resources} for lesson in lessons]
+            lessons_sizes_dict = {}
+
+            for lesson_node_item in lessons_nodes_list:
```

Comment:
If you don't do the above list comprehension, you can just do:
```
for lesson in lessons:
```

Also if you do this ^ then you don't even need the `if` statement, as iterating over an empty iterator will do no harm!